### ISSUE TITLE
Unable to build tripwire-2.4.3.1 with -Werror=terminate

### ISSUE THREAD

**Issue by nimiux** [Tripwire/tripwire-open-source#12]
More information at: https://bugs.gentoo.org/show_bug.cgi?id=595398
Thanks.

The specific error is:

```
x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H		  -I.. -I../..  -DCONFIG_DIR="\"/etc/tripwire\"" -fno-strict-aliasing  -Werror=terminate -march=native -O2 -pipe -c -o asn.o asn.cpp
In file included from asn.cpp:4:0:
asn.cpp: In destructor ‘virtual BERSequenceDecoder::~BERSequenceDecoder()’:
asn.h:15:45: error: throw will always call terminate() [-Werror=terminate]
 #define BERDecodeError() throw BERDecodeErr()
                                             ^
asn.cpp:90:13: note: in expansion of macro ‘BERDecodeError’
             BERDecodeError();
             ^~~~~~~~~~~~~~
asn.h:15:45: note: in C++11 destructors default to noexcept
 #define BERDecodeError() throw BERDecodeErr()
                                             ^
asn.cpp:90:13: note: in expansion of macro ‘BERDecodeError’
             BERDecodeError();
             ^~~~~~~~~~~~~~
```

**Comment by brc0x1** [comment]
Ok, I was able to repro this with gcc 6 and -Werror=terminate.  The build error here was the only error I encountered.  I should have a fix for this in 2.4.3.2, which ought to be wrapped up in the near future; right now I'm using noexcept(false), but I'll probably remove that and the throw-from-destructor, and just terminate/exit instead. That seems reasonable (I think) since the error in question would only happen if a keyfile is corrupt, which is unrecoverable anyway.
